### PR TITLE
Control.ts OP_CODES (0x61-0x69)

### PR DESCRIPTION
**Pull request by siradji**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:
Added OP_CODES 0x61-0x69

### CODE REVIEW

File: packages/jellyfish-transaction/src/script/control.ts [hunk 1]
```diff
@@ -8,3 +8,57 @@ export class OP_RETURN extends StaticCode {
     super(0x6a, 'OP_RETURN')
   }
 }
+
+export class OP_NOP extends StaticCode {
+  constructor () {
+    super(0x61, 'OP_NOP')
+  }
+}
```
Comment (fuxingloh) [inline packages/jellyfish-transaction/src/script/control.ts:16]: Add description for all, similar to how @jingyi2811 does it

**Review by fuxingloh — CHANGES_REQUESTED** [review]

File: packages/jellyfish-transaction/src/script/control.ts [hunk 2]
```diff
@@ -8,3 +8,57 @@ export class OP_RETURN extends StaticCode {
     super(0x6a, 'OP_RETURN')
   }
 }
+
+export class OP_NOP extends StaticCode {
```
Comment (jingyi2811) [inline packages/jellyfish-transaction/src/script/control.ts:12]: You should move ALL your code to the line before OP_RETURN.
Should be in this sequence : 61, 62, 63 .... 69, 6a

File: packages/jellyfish-transaction/src/script/mapping.ts [hunk 3]
```diff
@@ -343,15 +343,15 @@ export const OP_CODES = {
...
+  OP_NOP: new control.OP_NOP(),
+  OP_VER: new control.OP_VER(),
+  OP_IF: new control.OP_IF(),
+  OP_NOTIF: new control.OP_NOTIF(),
+  OP_VERIF: new control.OP_VERIF(),
+  OP_VERNOTIF: new control.OP_VERNOTIF(),
+  OP_ELSE: new control.OP_ELSE(),
+  OP_ENDIF: new control.OP_ENDIF(),
+  OP_VERIFY: new control.OP_VERIFY(),
   OP_RETURN: new control.OP_RETURN(),
```
Comment (jingyi2811) [inline packages/jellyfish-transaction/src/script/mapping.ts:355]: See: this is correct

File: packages/jellyfish-transaction/src/script/mapping.ts [hunk 4]
```diff
@@ -478,6 +478,15 @@ const HEX_MAPPING: {
   0x60: OP_CODES.OP_16,
   // control
   0x6a: OP_CODES.OP_RETURN,
```
Comment (jingyi2811) [inline packages/jellyfish-transaction/src/script/mapping.ts:480]: Apply to here too

**Comment by jingyi2811** [comment]
Remember to add #333 to your opcodes in https://github.com/DeFiCh/jellyfish/issues/107 when you are adding them.
Tick the checkboxes once this PR is accepted.

File: packages/jellyfish-transaction/src/script/mapping.ts (hunk 3 above)
Comment (siradji) [reply packages/jellyfish-transaction/src/script/mapping.ts:355]: Okay, i get it.

File: packages/jellyfish-transaction/src/script/control.ts (hunk 1 above)
Comment (siradji) [reply packages/jellyfish-transaction/src/script/control.ts:10]: I have added the descriptions. Let me know if i got it correctly.

File: packages/jellyfish-transaction/__tests__/script/mapping.test.ts [hunk 5]
```diff
@@ -153,4 +153,40 @@ describe('All mapped OP_CODES are setup properly: (static, hex, num, asm)', () =
...
+  it('OP_IF', () => {
+    expectOPCode(script.OP_CODES.OP_IF, script.OP_IF, 'OP_IF', 0X63, '63')
+  })
+
+  it('OP_NOTIF', () => {
+    expectOPCode(script.OP_CODES.OP_NOTIF, script.OP_NOTIF, 'OP_NOTIF', 0X64, '64')
+  })
+
+  it('OP_VERIF', () => {
+    expectOPCode(script.OP_CODES.OP_VERIF, script.OP_VERIF, 'OP_VERIF', 0X65, '65')
```
Comment (jingyi2811) [inline packages/jellyfish-transaction/__tests__/script/mapping.test.ts:174]: X should be lower case

File: packages/jellyfish-transaction/__tests__/script/mapping.test.ts [hunk 6]
```diff
@@ -153,4 +153,40 @@ describe('All mapped OP_CODES are setup properly: (static, hex, num, asm)', () =
...
+    expectOPCode(script.OP_CODES.OP_ELSE, script.OP_ELSE, 'OP_ELSE', 0X67, '67')
+  })
+
+  it('OP_ENDIF', () => {
+    expectOPCode(script.OP_CODES.OP_ENDIF, script.OP_ENDIF, 'OP_ENDIF', 0X68, '68')
+  })
+
+  it('OP_VERIFY', () => {
+    expectOPCode(script.OP_CODES.OP_VERIFY, script.OP_VERIFY, 'OP_VERIFY', 0X69, '69')
+  })
```
Comment (jingyi2811) [inline packages/jellyfish-transaction/__tests__/script/mapping.test.ts:191]: Move the OP_RETURN to the line after this line.

File: packages/jellyfish-transaction/src/script/mapping.ts [hunk 7]
```diff
@@ -478,6 +478,15 @@ const HEX_MAPPING: {
...
+  0x61: OP_CODES.OP_NOP,
+  0x62: OP_CODES.OP_VER,
+  0x63: OP_CODES.OP_IF,
+  0x64: OP_CODES.OP_NOTIF,
+  0x65: OP_CODES.OP_VERIF,
+  0x66: OP_CODES.OP_VERNOTIF,
+  0x67: OP_CODES.OP_ELSE,
+  0x68: OP_CODES.OP_ENDIF,
+  0x69: OP_CODES.OP_VERIFY,
```
Comment (jingyi2811) [inline packages/jellyfish-transaction/src/script/mapping.ts:489]: Move OP_RETURN to the line after this line

File: packages/jellyfish-transaction/src/script/control.ts [hunk 8]
```diff
@@ -1,5 +1,89 @@
...
+ * Puts the version of the protocol under which this transaction will be evaluated onto the stack.
+ */
+export class OP_VER extends StaticCode {
+  constructor () {
+    super(0x62, 'OP_VER')
+  }
+}
+
+/**
+ * If the top stack value is not False, the statements are executed
```
Comment (canonbrother) [inline packages/jellyfish-transaction/src/script/control.ts:22]: try not use negative statement
change to 
If ...... is True...

File: packages/jellyfish-transaction/src/script/control.ts [hunk 9]
```diff
@@ -1,5 +1,89 @@
...
+ */
+export class OP_IF extends StaticCode {
+  constructor () {
+    super(0x63, 'OP_IF')
+  }
+}
+
+/**
+ * If the top stack value is False, the statements are executed.
```
Comment (canonbrother) [inline packages/jellyfish-transaction/src/script/control.ts:31]: false

File: packages/jellyfish-transaction/__tests__/script/index.test.ts [hunk 10]
```diff
@@ -725,3 +725,165 @@ describe('P2WSH: [OP_0, OP_PUSHDATA<SHA256(script)>]', () => {
     expect(smartBuffer.toBuffer().toString('hex')).toStrictEqual(hex)
   })
 })
+
```
Comment (jingyi2811) [inline packages/jellyfish-transaction/__tests__/script/index.test.ts:728]: Move all these to the line after '0160' and before '016a'.

Comment (jingyi2811) [reply packages/jellyfish-transaction/__tests__/script/index.test.ts:728]: Also, check other sections too. It should be arranged in this order => 0160, 0161 ... 0169, 016a

**Review by jingyi2811 — APPROVED** [review]

**Review by canonbrother — APPROVED** [review]

**Review by fuxingloh — APPROVED** [review]

**Pull request merged**